lib/ansible/modules/cloud/amazon/ec2_customer_gateway_facts.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+---
+module: ec2_customer_gateway_facts
+short_description: Gather facts about customer gateways in AWS
+description:
+    - Gather facts about customer gateways in AWS
+version_added: "2.5"
+requirements: [ boto3 ]
+author: Madhura Naniwadekar(@Madhura-CSI)
+options:
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value.
+        See U(http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeCustomerGateways.html) for possible filters.
+    required: false
+    default: None
+  customer_gateway_ids:
+    description:
+      - Get details of a specific customer gateways using customer gateway ID/IDs. This value should be provided as a list.
+    required: false
+    default: None
+extends_documentation_fragment:
+    - aws
+    - ec2
+
+'''
+
+EXAMPLES = '''
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather facts about all customer gateways
+  ec2_customer_gateway_facts:
+
+- name: Gather facts about a filtered list of customer gateways, based on tags
+  ec2_customer_gateway_facts:
+    region: ap-southeast-2
+    filters:
+      "tag:Name": test-customer-gateway
+      "tag:AltName": test-customer-gateway-alt
+  register: cust_gw_facts
+
+- name: Gather facts abouit a specific customer gateway by specifying customer gateway ID
+  ec2_customer_gateway_facts:
+    region: ap-southeast-2
+    customer_gateway_ids: ['cgw-fec843ce', 'cgw-fec844ce']
+  register: cust_gw_facts
+'''
+
+RETURN = '''
+CustomerGateways:
+    description: List of one or more customer gateways.
+    returned: always
+    type: list
+    sample: [
+            {
+                "bgp_asn": "65000",
+                "customer_gateway_id": "cgw-fec844ce",
+                "customer_gateway_name": "test-customer-gw",
+                "ip_address": "110.112.113.120",
+                "state": "available",
+                "tags": [
+                    {
+                        "key": "Name",
+                        "value": "test-customer-gw"
+                    }
+                ],
+                "type": "ipsec.1"
+            }
+        ]
+changed:
```

Comment:
No need to document `changed` in `RETURN` - it's widely understood. Also, it's never `True` in this module so the docs are incorrect!

Reply:
Addressed this change with latest commits